lib/fog/aws/models/storage/file.rb

```diff
@@ -328,18 +328,34 @@ def multipart_save(options)
           # Store ETags of upload parts
           part_tags = []
 
+          # Calculate total size and ensure we don't exceed part limit
+          total_size = Fog::Storage.get_body_size(body)
+          parts_count = (total_size.to_f / multipart_chunk_size).ceil
+
+          # AWS S3 has a hard limit of 10,000 parts, make sure we are below this limit for large objects
+          if parts_count > 10000
```

Comment:
This is a AWS limit but some other providers have lower limits, eg: Scaleway has a 1000 parts limit https://www.scaleway.com/en/docs/object-storage/faq/#how-can-i-upload-large-objects

It would be awesome if the maximum value was a configurable value, like they did in rclone: https://github.com/rclone/rclone/commit/7458d37d2a083d53f7d3847fcd8e581661ce63b7

Reply:
Good callout. I'd welcome a PR that makes this adjustable if you'd like to take a shot at it. Thanks!